common/ts/prepare-task.ts

```diff
@@ -37,6 +37,14 @@ export default async function prepareTask(endpoint: Endpoint, rootPath: string)
   await scanner.runPrepare();
 }
 
+function branchFeatureSupported(endpoint) {
+  if (endpoint.type === EndpointType.SonarCloud) {
+    return true;
+  }
+  // TODO check SQ version >= 7.2 and commercial branch feature available
```

Comment:
Does this mean that sonarqube extension will start requiring the "commercial branch feature" for pull request comment integrations with once version 4 of the plugins supports it?

Reply:
@Daniel-Svensson we are still discussing the options, but our current plan is:
* don't change the behavior when SQ < 7.2 = don't try to pass any parameter.
* when we detect SQ >= 7.2, pass automatically all sonar.branch.* or sonar.pullrequest.* parameters that are needed to have proper branch/PR decoration support. But you're right, this depends on the commercial feature.

Is it a problem for you?

Reply:
Currently we use sonarqube for the pull request comment support (as described on https://docs.sonarqube.org/display/SCAN/SonarQube+Extension+3.0) using the free version.
This depends on "sonar.analysis.mode=preview" as I understand it, will that possibility be removed in 7.2 ? It is a bit sad too loose such a feature when upgrading the vsts extension from v3.

While we have discussed uppgrading and maybe bying developer edition that is a process which is not very fast (to get approval). Loosing pr commenting might have the effect of stopping such effort. 

Continue making a nice Product. Features such as 2-way integrationg (changing status etc) between VSTS comments and sonarqube server would definitly help motivate paying for the product